x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java

```diff
@@ -28,13 +28,13 @@ public ParsingException(String message, Exception cause, int line, int charPosit
     public ParsingException(Source source, String message, Object... args) {
         super(message, args);
         this.line = source.source().getLineNumber();
-        this.charPositionInLine = source.source().getColumnNumber();
+        this.charPositionInLine = source.source().getColumnNumber() - 1;
```

Comment:
`Location#getColumnNumber()` increments the count by 1 already, which this class also does in its own `getColumnNumber()` -- the bug.

Reply:
This is interesting find, we are really off by one, I have never noticed it 😂 ! It worths a comment.

I wonder if it is a good idea to replace the three math with one here, and keep the column number from source unchanged? 

```
    public ParsingException(String message, Exception cause, int line, int charPositionInLine) {
        super(message, cause);
        this.line = line;
        this.charPositionInLine = charPositionInLine + 1;
    }

    ParsingException(String message, Object... args) {
        this(Source.EMPTY, message, args);
    }

    public ParsingException(Source source, String message, Object... args) {
        super(message, args);
        this.line = source.source().getLineNumber();
        this.charPositionInLine = source.source().getColumnNumber();
    }

    public ParsingException(Exception cause, Source source, String message, Object... args) {
        super(cause, message, args);
        this.line = source.source().getLineNumber();
        this.charPositionInLine = source.source().getColumnNumber();
    }

    public int getLineNumber() {
        return line;
    }

    public int getColumnNumber() {
        return charPositionInLine;
    }
```

Reply:
Indeed, thanks!